R/draw-dm.R

```diff
@@ -334,3 +325,76 @@ dm_get_colors <- function(dm) {
 dm_get_available_colors <- function() {
   c("default", colors())
 }
+
+
+#' Add info about a dm's tables
+#'
+#' @inheritParams dm_draw
+#' @param ... Descriptions for tables to set in the form `description = table`.
+#' These descriptions will be displayed when creating a diagram with [dm_draw()].
+#' Multi-line descriptions can be achieved using the newline symbol `\n`.
+#' The currently set descriptions can be checked using [dm_get_table_description()].
+#' Descriptions can be removed using `NULL = table`.
+#'
+#' `tidyselect` is supported, see [dplyr::select()] for details on the semantics.
+#'
+#' @return A `dm` object containing descriptions for specified tables.
+#' @export
+#'
+#' @examples
+dm_set_table_description <- function(dm, ...) {
+  check_not_zoomed(dm)
+  selected <- eval_select_table(quo(c(...)), src_tbls_impl(dm), unique = FALSE)
+  def <- dm_get_def(dm, quiet = TRUE)
+  reduce2(
+    selected,
+    names(selected),
+    function(def, table, desc) {
+      # `eval_select_table()` interprets the name `NULL` as "NULL"
+      # downside is, that no table description can be defined as "NULL"
+      desc <- if (desc == "NULL") {
+        NULL
+      } else {
+        desc
+      }
+      def$data[[which(def$table == table)]] <- structure(
+        def$data[[which(def$table == table)]],
+        "description" = desc
```

Comment:
```suggestion
        "label" = desc
```

This might be one of the cases where a `for` loop is easier to decipher. Also, `eval_select_both()` .

Reply:
We now can use the new functions from {labelled}.